src/ast/passes/semantic_analyser.cpp

```diff
@@ -3262,15 +3262,14 @@ void SemanticAnalyser::visit(AttachPoint &ap)
         has_end_probe_ = true;
       }
     }
-  } else if (ap.provider == "fentry" || ap.provider == "fexit") {
-    if (!bpftrace_.feature_->has_fentry()) {
-      LOG(ERROR, ap.loc, err_)
-          << "fentry/fexit not available for your kernel version.";
+  } else if (ap.provider == "bpftrace") {
+    if (ap.target == "signal") {
+      if (SIGNALS.find(ap.func) == SIGNALS.end())
+        LOG(ERROR, ap.loc, err_) << ap.func << " is not a supported signal";
       return;
     }
-
-    if (ap.func == "")
-      LOG(ERROR, ap.loc, err_) << "fentry should specify a function";
```

Comment:
Looks like #3524 caused this duplicated code.

Will need a rebase past #3538. Let's do a clean fix to intent is saved into git.